packages/mystmd-py/src/mystmd_py/main.py

```diff
@@ -4,41 +4,87 @@
 import subprocess
 import sys
 import re
+import platformdirs
 
 
-def main():
-    NODE_LOCATION = (
-        shutil.which("node") or shutil.which("node.exe") or shutil.which("node.cmd")
+NODEENV_VERSION = "18.0.0"
+
+
+def find_installed_node():
+    return shutil.which("node") or shutil.which("node.exe") or shutil.which("node.cmd")
+
+
+def find_nodeenv_path():
+    return platformdirs.user_data_path(
+        appname="myst", appauthor=False, version=NODEENV_VERSION
     )
-    PATH_TO_BIN_JS = (pathlib.Path(__file__).parent / "myst.cjs").resolve()
 
-    if not NODE_LOCATION:
+
+def create_nodeenv(env_path):
+    command = [
+        sys.executable,
+        "-m",
+        "nodeenv",
+        "-v",
+        f"--node={NODEENV_VERSION}",
+        "--prebuilt",
+        "--clean-src",
+        env_path,
+    ]
+    subprocess.run(command, check=True)
+
+
+def find_any_node(binary_path):
+    node_path = find_installed_node()
+    if node_path is not None:
+        return pathlib.Path(node_path).absolute(), binary_path
+
+    nodeenv_path = find_nodeenv_path()
+    if not nodeenv_path.exists():
+        print(f"🔍 Couldn't find installed `node`.\n\n⚙️ Installing Node.js in {nodeenv_path}")
+        create_nodeenv(nodeenv_path)
```

Comment:
We should give the user control here - "Would you like to install node now using nodeenv?" y/n